src/Services/QuestionnaireResponseService.php

```diff
@@ -325,93 +328,117 @@ public function parseQuestionnaireResponseForms($response): array
         return $question_results;
     }
 
-    /* WIP */
     /**
-     * @param $pid
-     * @param $foreign_id
-     * @param $name
-     * @param $q
-     * @param $response
-     * @param $form_response
+     * @param       $response
+     * @param       $pid
+     * @param null  $encounter
+     * @param null  $qr_id
+     * @param null  $qr_record_id
+     * @param null  $q
+     * @param null  $q_id
+     * @param null  $form_response
+     * @param array $scores
      * @return array|false|int|mixed
+     * @throws Exception
      */
-    public function saveQuestionnaireResponse($pid, $foreign_id, $name, $q, $response, $form_response = null)
-    {
+    public function saveQuestionnaireResponse(
```

Comment:
@bradymiller and @adunsulag 
You guys can beat me up on these classes if you want to advise.
I keep wanting to over engineer for future expectations so you'll see some redundancies but I think everything here is okay so far.